src/httpClient/dumpClient.ts

```diff
@@ -18,16 +18,8 @@ export class DumpServerClient extends BaseClient {
     super(logger);
   }
 
-  public async postDumpMetadata(dumpServerConfig: Required<DumpServerConfig>, body: DumpMetadataCreationBody): Promise<HttpResponse<string>> {
-    const { dumpServerEndpoint: endpoint, dumpServerHeaders: headers } = dumpServerConfig;
-
-    const requestHeaders: Record<string, string> = {};
-    headers.map((headerKeyValue) => {
-      const [key, value] = headerKeyValue.trim().split('=');
-      requestHeaders[key] = value;
-    });
-
-    this.logger.info({ msg: 'invoking POST http request', url: `${endpoint}/${DUMP_METADATA_ENDPOINT}`, headers: Object.keys(requestHeaders) });
+  public async postDumpMetadata(endpoint: string, body: DumpMetadataCreationBody, headers?: Record<string, string>): Promise<HttpResponse<string>> {
+    this.logger.info({ msg: 'invoking POST http request', url: `${endpoint}/${DUMP_METADATA_ENDPOINT}`, headers: Object.keys(headers ?? {}) });
```

Comment:
those are only the user supplied headers, not all the headers, and not sure there is any value in listing only the keys.